File: For Fun/Assignment 2/C# version/CountDigits.cs
using System;

class CountDigits
{
	public static void Main(string[] args)
	{
		Console.Write("Provide a positive integer: ");
		int input = Convert.ToInt32(Console.ReadLine());

		int odds = 0;
		int zeros = 0;
		int evens = 0;

		while (input != 0)
		{
			int temp = input % 10;

			if (temp == 0)
			{
				zeros++;
			}
			else if (temp % 2 == 0)
			{
				evens++;
			}
			else
			{
				odds++;
			}
			input /= 10;
		}

		Console.WriteLine("Zeros: " + zeros);
		Console.WriteLine("Odd: " + odds);
		Console.WriteLine("Even: " + evens);
	}
}

File: sb223ce_assign2/CountDigits.java
package sb223ce_assign2;

import java.util.Scanner;

public class CountDigits {

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);

		/* User instruction */
		System.out.print("Provide a positive integer: ");
		int sum = scan.nextInt();

		/* We need 'while' statement because we need to run the loop until its
		 * zero and we cannot find the length of integer. Then, we will take one
		 * integer at one time through 'if' statements.*/

		int zero = 0;
		int even = 0;
		int odd = 0;

		while (sum != 0) {
			int digit = sum % 10; /* Pick last digit */

			if (digit == 0) /*For counting zeros separately*/ 
			{
				zero++;
			} else if (digit % 2 == 0)  /* Even numbers have 0 mode with 2 */
			{
				even++;
			} else {
				odd++;
			}

			sum = sum / 10; /* Remove last digit */
		}

		System.out.println("Even: " + even);
		System.out.println("Odd: " + odd);
		System.out.println("Zeros: " + zero);

		scan.close();

	}
}
